lib/core/rewriter/rewriter_fabric.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "core/rewriter/rewriter_proxy.h"
+
+namespace clang {
+class SourceManager;
+class LangOptions;
+}  // namespace clang
+namespace oklt {
+enum class RewriterProxyType {
+    Original,
+    WithDeltaTree,
+};
+
+std::unique_ptr<RewriterProxy> makeRewriterProxy(clang::SourceManager&,
```

Comment:
I'm not sure. But I suspect that DeltaTree proxy is injected for certain scenarios.
It can be a runtime option when creating Rewriter, or compile time via Template argument.

Reply:
I know that template argument would remove virtual calls, but then we will have to copy a ton of code between `EmptyRewriterProxy` and `DtreeRewriterProxy` since 80% of the code is the same between them (which is currently implemented in parent `RewriterProxy`

Reply:
I see. OK.